model/offline/JsonOfflineTestParser.php

```diff
@@ -76,8 +89,60 @@ protected function getContent()
         if (!$this->content) {
             $this->content = $this->getFile()->read();
         }
+
         return $this->content;
+    }
+
+    /**
+     * It checks if URI belongs the instance
+     *
+     * @param $uri
+     *
+     * @return bool
+     */
+    public function checkUri($uri)
+    {
+        /** @var UriProvider $uriProvider */
+        $uriProvider = $this->getUriProviderService();
+        $validInstanceLink = $uriProvider->provide();
+
+//        print_r($uri);
+//        echo '    ';
+//        print_r($validInstanceLink);
+//        echo PHP_EOL;
+
+        return Uri::getDomain($uri) === Uri::getDomain($validInstanceLink);
+    }
+
+    /**
+     * Validates body content to avoid unexpected behavior
+     */
+    public function validateUris()
+    {
+        $body = $this->getBody();
+
+        $validateUriFunction = [$this, 'checkUri'];
 
+        $invalid = 0;
+
+        array_walk_recursive($body, static function ($item, $key) use ($validateUriFunction, &$invalid) {
+            if (in_array($key, self::PROTECTED_FIELDS, true) && !$validateUriFunction($item)) {
```

Comment:
this algorithm doesn't take to account nesting level of a field. is it OK?

Reply:
In docs said "Applies the user-defined callback function to each element of the array. This function will recurse into deeper arrays." that means that this function will go to nested levels.
https://www.php.net/manual/en/function.array-walk-recursive.php